core/controllers/features.py

```diff
@@ -27,6 +27,22 @@ class ExplorationFeaturesHandler(base.BaseHandler):
     """Returns features the given exploration is configured to support."""
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'exploration_id': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'has_length_at_most',
+                    'max_value': 12 # Max length of an exploration_id
+                }, {
+                    'id': 'is_nonempty'
+                }]
```

Comment:
Can we introduce regex instead?

Reply:
I have looked into the codebase and it seems that from [here exp_id's](https://github.com/oppia/oppia/blob/d2fdf35a3c4727315ef19031b585fe3209baef9c/core/storage/base_model/gae_models.py#L380) are generated which is a random hash value within a particular range.
Thus, is there any pattern which will be captured by regex ?
If yes, can you please help me find one.

Reply:
It is always base64 encoded, except that we replace '=' with 'c' (https://github.com/oppia/oppia/blob/d2fdf35a3c4727315ef19031b585fe3209baef9c/utils.py#L462) so you can use these chars: https://stackoverflow.com/a/13195218

Reply:
I have added a regex, please take a look.
Thank you.

Reply:
Sorry, I just went through it and we might actually need to allow `-` and `_` instead of `+` and `/`, since Python uses the URL and Filename safe alphabet https://www.rfc-editor.org/rfc/rfc4648.txt (page 7).

Reply:
Also since we know the max length of the ID you can use `^[a-zA-Z0-9\-_]{1,12}$`. Also, should we set this regex as a constant somewhere?

Reply:
Updated regex.
Added this constant in feconf.py, please take a look.